docs/organization-integration/onpremise-kubernetes.md

```diff
@@ -0,0 +1,444 @@
+database access credentials via a Kubernetes secret. One way to configure this is as follows:
+
+```yaml
+auth-api:
+  config:
+    datasource:
+      data:
+        url: "some.host/sigridauthdb"
+        username: "webapp-user"
+        password: "S3cr3t"
+```
+
+The above configuration results in the Helm chart rendering a Kubernetes secret containing the
+URL, username and password. Obviously, a `values.yaml` should not contain a cleartext password, 
+typically setting the password interactively using the Helm CLI is preferred 
+`helm install <name> <chart> --set auth-api.config.datasource.data.password="S3cr3t"`).
+
+The other option is to create a Kubernetes secret elsewhere and reference it. Suppose the name of
+this secret is `postgresql-sigridauthdb`, then the configuration is:
+
+```yaml
+auth-api:
+  config:
+    datasource:
+      create: false
+      secretName: postgresql-sigridauthdb
+```
+
+## (D) Identity provider
+
+The on-premise version of Sigrid only supports single-sign-on (SSO); it does not maintain users 
+by itself. Hence, a connection to an identity provider (IdP) is mandatory and needs to be 
+configured in the Helm chart.
+
+The only supported protocol is OpenID Connect and consequently, it is best if an IdP compatible 
+with OpenID Connect is available (e.g., [Okta](https://www.okta.com) or [Auth0](https://auth0.com)). 
+While [Microsoft Entra](https://www.microsoft.com/en-us/security/business/microsoft-entra)
+also supports OpenID Connect, it is currently not possible to connect to Microsoft Entra directly. 
+Instead, a bridge needs to be used, such as [Keycloak](https://www.keycloak.org) or
+[Dex](https://dexidp.io). The latter is provided by Sigrid's Helm chart as an option.
+
+Note that for on-premise deployments, using SAML is only possible via a bridge such as [Dex]
+(https://dexidp.io). Consequently, [SIG's documentation](usermanagement.md#2-using-single-sign-on-sso-with-an-identity-provider-idp) for connecting hosted Sigrid to an IdP 
+does not apply.
+
+Configuring an identity provider (IdP) is a two-step process:
+1. Register an OpenID Connect client in the IdP.
+2. Configure the client in Sigrid's Helm chart.
+
+### Step 1: Register an OpenID Connect (OIDC) client in the IdP
+
+We refer to the documentation of the IdP on how to register an OIDC client/app. This is often 
+called "Add an OIDC application" or "Create an OIDC integration" in the documentation.
+
+You typically need to provide a redirect URI (also called "login callback URL"). For on-premise 
+Sigrid, this is _always_ `https://YOUR-DOMAIN.COM/rest/auth/login/oauth2/code/sigridmfa`, where 
+`YOUR_DOMAIN.COM` is a placeholder for the (sub)domain on which the on-premise deployment of 
+Sigrid will be hosted and configured as described in Section B in this document.
+
+Registering an OpenID Connect client/app results in a client ID and client secret, which are 
+generated by the IdP. We'll need those in the next step.
+
```

Comment:
This was not in the original; it came to light yesterday that this is a requirement so I suggest we add it.
```suggestion

Note that many IdPs allow configuration of the information provided to clients in the OIDC identity token. Sigrid expects that this token reveals the email address, first name and last name of the user.

```